File: docs/APIs-and-SDKs/SDK-Documentation/complete-example/dotnet/completeExample.cs
using ABSmartly;
using Microsoft.Extensions.DependencyInjection;

// Initialize the SDK
var serviceProvider = new ServiceCollection()
    .AddHttpClient()
    .BuildServiceProvider();

var httpClientFactory = serviceProvider
    .GetService<IHttpClientFactory>();

var abSdk = new ABSdk(
    new ABSdkHttpClientFactory(httpClientFactory),
    new ABSmartlyServiceConfiguration
    {
        Environment = "production",
        Application = "website",
        Endpoint = "https://your-company.absmartly.io/v1",
        ApiKey = "YOUR-API-KEY"
    });

// Create a context for this user
var config = new ContextConfig()
    .SetUnit("user_id", "user-12345");

// Using the disposable pattern ensures events are flushed on close
using var context = abSdk.CreateContext(config);
await context.WaitUntilReady();

// Check which variant the user is in
if (context.GetTreatment("homepage_banner_experiment") == 0)
{
    // Variant A (control): show the existing banner
    ShowBanner("Welcome back!");
}
else
{
    // Variant B: show a personalized banner
    ShowBanner("Welcome back, we have new deals for you!");
}

// Use a variable set in the Web Console for more flexibility
var buttonColor = context.GetVariableValue("button.color", "blue");
SetButtonColor(buttonColor);

// Track when the user completes a key action
context.Track("cta_click", new Dictionary<string, object>
{
    { "page", "homepage" }
});

// Track a purchase with revenue data
context.Track("purchase", new Dictionary<string, object>
{
    { "revenue", 49.99 },
    { "item_count", 3 }
});
// Context is automatically closed and flushed by the using statement

File: docs/APIs-and-SDKs/SDK-Documentation/complete-example/ruby/completeExample.rb
require "absmartly"

# Initialize the SDK
Absmartly.configure_client do |config|
  config.endpoint = "https://your-company.absmartly.io/v1"
  config.api_key = "YOUR-API-KEY"
  config.application = "website"
  config.environment = "production"
end

# Create a context for this user
context_config = Absmartly.create_context_config
context_config.units = { user_id: "user-12345" }
context = Absmartly.create_context(context_config)
context.wait_until_ready

# Check which variant the user is in
treatment = context.treatment("homepage_banner_experiment")

if treatment == 1
  # Variant B: show a personalized banner
  show_banner("Welcome back, we have new deals for you!")
else
  # Variant A (control): show the existing banner
  show_banner("Welcome back!")
end

# Use a variable set in the Web Console for more flexibility
button_color = context.variable_value("button.color", "blue")
set_button_color(button_color)

# Track when the user completes a key action
context.track("cta_click", { page: "homepage" })

# Track a purchase with revenue data
context.track("purchase", {
  revenue: 49.99,
  item_count: 3,
})

# Close the context to flush remaining events
context.close
